kubernetes/client.go

```diff
@@ -79,6 +79,10 @@ type IstioClient struct {
 	k8s                *kube.Clientset
 	istioConfigApi     *rest.RESTClient
 	istioNetworkingApi *rest.RESTClient
+	// isOpenShift can be cached per client
+	isOpenShift *bool
```

Comment:
could it be just "bool" ? (i don't know, it seems weird to get a pointer just for a boolean - what's the reason?)

Reply:
I wanted to represent 3 states: not defined, true or false.
So, if it's not defined, it means that kiali didn't query if we are in kubernetes or openshift.
Returning a false in all cases could be misleading.

Reply:
ah, ok

Reply:
although, it seems more typical in Go to return two values in that case.

Reply:
I agree that Go typically returns two values for this kind of thing (IsOpenShift() (bool err)).
If this code remains using a pointer, I highly recommend you put a comment above this to explain why it is a pointer and that a nil is expected when not set.

Reply:
This code is going to use a pointer as it represents three states and no two.